pkg/extensions/sync/on_demand.go

```diff
@@ -125,13 +132,28 @@ func OneImage(cfg Config, storeController storage.StoreController,
 		} else {
 			log.Info().Msgf("successfully synced %s", upstreamRef.DockerReference().Name())
 
-			err := pushSyncedLocalImage(repo, tag, uuid.String(), storeController, log)
-			if err != nil {
+			if err = retry.RetryIfNecessary(context.Background(), func() error {
```

Comment:
what is this doing?

Reply:
will remove it, added it by mistake, thanks for spotting it.